### DISCUSSION TITLE
Ideas and features

### DISCUSSION THREAD

**Discussion by GiorgosXou** [GiorgosXou/TUIFIManager#78] [Ideas]
Just because I don't have a well centralised place to keep my TUIFI ideas at the momment, I decided to create this place where i will post some potential ideas

**Comment by GiorgosXou** [comment]
💡 [Create a watchdog in Python to look for filesystem changes](https://www.geeksforgeeks.org/create-a-watchdog-in-python-to-look-for-filesystem-changes/) *(with an option for those who don't want to use it [for optimization reasons])*

**Comment by GiorgosXou** [comment]
💡 Auto clone repositories and download files if theres any link in the clipboard 
- *(reminder to check if also files present on the "virtual clipboard" of tuifi)* 
- *(maybe discard virtual clipboard on  the change of the real clipboard ?)*

**Comment by Sigmanificient** [comment]
I think using discussion for this could be better?

**Reply by GiorgosXou** [reply]
Fixed it xD

**Comment by GiorgosXou** [comment]
:bulb: individual thread with progressbar when copying files *(opperations on large files is slow sometimes)*

**Comment by GiorgosXou** [comment]
:bulb: Focus on file when `tuifi` is called with a file *(For apps that have an option of "show in folder" eg. chromium under downloads)*

**Comment by GiorgosXou** [comment]
:bulb: Navigation history *(`CTRL+O`\\`H` `CTRL+I`\\`L`   [although ctrl+h seems to output something weird, anyways] and something for the normal mode)* + TUI arrow-buttons on the bottom bar?

**Comment by GiorgosXou** [comment]
💡 TabStates - a memory efficient tab-component that uses a single instance of `TUIFIManager`

**Comment by GiorgosXou** [comment]
TODO: Add Icon 
```
    '/stl':TUIFIProfile((
        ' ╭─╭┰─╮─╮ \n'
        ' │╭╰┸─╯╮│ \n'
        ' │╰.STL╯│ \n'
        ' ╰━━━━━━╯ '
    ),4, DEFAULT_OPENER),
    '/gcode':TUIFIProfile((
        ' ╭─╭─┰╮─┒ \n'
        ' │[╰┬┸╯]┇ \n'
        ' │GCODE:┃ \n'
        ' ╰━━━━━━┛ '
    ),1, DEFAULT_OPENER),
    '/fcstd':TUIFIProfile((
        ' ┏━━━━━━┓ \n'
        ' ┃░█▀▀/:┃ \n'
        ' ┃░█▀CAD┇ \n'
        ' ┗━━━━━━┛ '
    ),3, DEFAULT_EDITOR),
    '/lock':TUIFIProfile((
        ' ┏┏━━━━┓┓ \n'
        ' ┣┗LOCK┛┫ \n'
        ' ┃┋┇[]┇┋┃ \n'
        ' ┗━━━━━━┛ '
    ),2, DEFAULT_EDITOR),

```

**Comment by GiorgosXou** [comment]
:bulb:  left-right treeview *(with pined locations too)*

**Comment by GiorgosXou** [comment]
💡 Recent files

**Comment by GiorgosXou** [comment]
💡 Plugins for more complex things *(eg. ability to integrate [`z`](https://github.com/rupa/z) command)*

**Comment by GiorgosXou** [comment]
💡 When navigating backwords\down focus on the folder you were at

**Reply by GiorgosXou** [reply]
https://github.com/GiorgosXou/TUIFIManager/commit/8696c3630f5c9ec0b8479da3604cf4562abe5dfd

**Comment by GiorgosXou** [comment]
💡 Add extensions: `iml` `dart`, `swift` `cmake`

**Comment by GiorgosXou** [comment]
💡 Add custom right click options per file

**Reply by GiorgosXou** [reply]
like zip

**Comment by GiorgosXou** [comment]
💡 Corrupted or 0-byte files like `psd` to get other icon or something?

**Reply by GiorgosXou** [reply]
eg. thunar does this 
![image](https://github.com/GiorgosXou/TUIFIManager/assets/11987271/b25e3ed5-f45a-4011-b2a6-e7c5521bb84e)

**Comment by GiorgosXou** [comment]
💡 Icon packs

**Reply by GiorgosXou** [reply]
But i think I have to fix color support stuff first

**Comment by GiorgosXou** [comment]
Random potential Icon Ideas:
```
    '/7z':TUIFIProfile((
        ' ┳━━┫┣━┓┳ \n'
        ' ┃┃░┃┣━━┫ \n'
        ' ┃┃░┃┣━━┫ \n'
        ' ┻━━┫┣━┛┻ '
    ),2, DEFAULT_OPENER),
    '/zip':TUIFIProfile((
        ' ┳━━┳┣━┓┳ \n'
        ' ┻┃░┃┣━━┫ \n'
        ' ┳┃░┃┣━━┫ \n'
        ' ┻━━┻┣━┛┻ '
    ),2, DEFAULT_OPENER),
    '/lz':TUIFIProfile((
        ' ╭━━┓┣━╮┳ \n'
        ' ┃┃░┃┣━━┫ \n'
        ' ┃┃░┃┣━━┫ \n'
        ' ╰━━┫┣━╯┻ '
    ),2, DEFAULT_OPENER),

```

**Reply by GiorgosXou** [reply]
That's a nice one
```
    '/zip':TUIFIProfile((
        ' ▃▃▃│╮▃▃▃ \n'
        ' ▒▒▒││▒▒▒ \n'
        ' ▓▓▓││▓▓▓ \n'
        ' ZIP╰│▀▀▀ '
    ),2, DEFAULT_OPENER),
```

**Reply by GiorgosXou** [reply]
Ah.. this feels so niceeee
![image](https://github.com/GiorgosXou/TUIFIManager/assets/11987271/c3dfb1d5-c0a9-4e3d-be6c-feb7182e2d14)

**Reply by Sigmanificient** [reply]
I think allowing to have a second tone for the text info (ZIP in logo, PNG, ...) be could nice, what do you think?

**Reply by GiorgosXou** [reply]
[I think that every character could have a different tone, but it's not the right time yet](https://github.com/GiorgosXou/TUIFIManager/blob/055522f6b71d89ae27fa4247737507ed77cf32ab/TUIFIManager/TUIFIProfile.py#L12 "check line 12"). I might also drop support for very old non 256-colored terminals *(which will fix the [theme issue](https://github.com/GiorgosXou/TUIFIManager/issues/38#issuecomment-1377731164) too)*.
